qiskit/algorithms/gradients/base_sampler_gradient.py

```diff
@@ -134,3 +137,16 @@ def _validate_arguments(
                     f"The number of values ({len(parameter_value)}) does not match "
                     f"the number of parameters ({circuit.num_parameters}) for the {i}-th circuit."
                 )
+
+    def _update_result_run_options(self, run_options: dict) -> dict:
```

Comment:
Since this function returns a copy of run options for this execuction, would `_get_local_run_options` be a more precise name?

Reply:
Thank you for reviewing. I changed the func name to `_get_local_run_options`!